includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -222,6 +222,20 @@ public function __construct() {
 		add_action( 'set_logged_in_cookie', [ $this, 'set_cookie_on_current_request' ] );
 
 		add_action( 'wc_ajax_wc_stripe_save_appearance', [ $this, 'save_appearance_ajax' ] );
+
+		add_filter( 'woocommerce_saved_payment_methods_list', [ $this, 'filter_saved_payment_methods_list' ], 10, 2 );
+	}
+
+	/**
+	 * Removes all saved payment methods when the setting to save cards is disabled.
+	 *
+	 * @return array An empty list of payment methods
+	 */
+	public function filter_saved_payment_methods_list( $item, $payment_token ) {
+		if ( ! $this->saved_cards ) {
+			return [];
+		}
+		return $item;
 	}
```

Comment:
I think the second param is supposed to be `customer_id`, it's not used, but suggesting updating the docs anyway 🤷🏼  

https://woocommerce.github.io/code-reference/files/woocommerce-includes-wc-account-functions.html:L345-L381

```suggestion
	/**
	 * Removes all saved payment methods when the setting to save cards is disabled.
	 *
	 * @param  array $list         List of payment methods passed from wc_get_customer_saved_methods_list().
	 * @param  int   $customer_id  The customer to fetch payment methods for.
	 * @return array               Filtered list of customers payment methods.
	 */
	public function filter_saved_payment_methods_list( $list, $customer_id ) {
		if ( ! $this->saved_cards ) {
			return [];
		}
		return $list;
	}
```